tools/autograph-client/build_test_xpis.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env sh
```

Comment:
Did you test if the zips get verified by Firefox again?
For Firefox unit testing it would be good to have (in addition to xpis in here) test cases with intermediate certs and one with RSA only.

Reply:
Haven't tested against Fx since https://bugzilla.mozilla.org/show_bug.cgi?id=1466683, which was against an earlier commit on this branch. I'll add the other test cases.

Reply:
Actually, I'm not sure what test cases to add. Multiple intermediate certs? An ecdsa intermediate (I don't think we can do this)? For the RSA only one, do you mean just a pk7 signature and COSE PS256?

Reply:
We support using intermediate certs or the root directly when verifying. I want to make sure that we test both (if you support signing with the root that is :). It doesn't matter what keys the intermediates use. We're only restricted in what key is used in the COSE signature.

> For the RSA only one, do you mean just a pk7 signature and COSE PS256?

Yep

Reply:
OK I added the pk7, PS256 test case here: https://github.com/mozilla-services/autograph/pull/76/commits/238d19f93424c2207d5536fa6b8d1303d320f534#diff-c93506e8ef6730d1fc392b838221a31dR20

We only support one intermediate and don't sign with the root.

So I think we're only missing tests against an ECDSA intermediate with an ECDSA pk7 EE (we decided not to support generating pk7 EEs that don't match the intermediate private key type) and the various COSE options (just pk7, ES256, PS256, ES256 and ES384, ES256 and PS256).

I don't think we have an ECDSA intermediate in the stage or prod configs, so I'll have to find someone with access to the roots to generate those.

Reply:
Testing the roots and multiple intermediates might be implemented as an optional "COSE message key id prefix" param taking multiple args.